From f397d61593ca447926c3fb90a10d3e83a7d0ac6e Mon Sep 17 00:00:00 2001
From: Benjamin Drung <benjamin.drung@profitbricks.com>
Date: Mon, 29 May 2017 16:50:23 +0200
Subject: [PATCH 2/3] Install fastnetmon.service with cmake

The path to the fastnetmon daemon might differ, thus use configure_file
to set the path dynamically.
Forwarded: yes, https://github.com/pavel-odintsov/fastnetmon/pull/663
---
 src/CMakeLists.txt                                | 7 +++++++
 src/{fastnetmon.service => fastnetmon.service.in} | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)
 rename src/{fastnetmon.service => fastnetmon.service.in} (85%)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,6 +27,9 @@
 set (FASTNETMON_VERSION_MAJOR 1)
 set (FASTNETMON_VERSION_MINOR 1)
 
+set(CMAKE_INSTALL_SYSTEMD_SERVICEDIR "/lib/systemd/system"
+  CACHE PATH "Location for systemd service files")
+
 if (ENABLE_GOBGP_SUPPORT)
     # We could not compile gRPC without C++ 11
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -std=c++11")
@@ -600,6 +603,10 @@
 install(FILES man/fastnetmon.1 DESTINATION /usr/share/man/man1)
 install(FILES man/fastnetmon_client.1 DESTINATION /usr/share/man/man1)
 
+# service files
+configure_file(fastnetmon.service.in "${CMAKE_CURRENT_BINARY_DIR}/fastnetmon.service" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/fastnetmon.service" DESTINATION ${CMAKE_INSTALL_SYSTEMD_SERVICEDIR})
+
 # Configure cpack package builder
 # Run it with: cd build; cpack -G DEB ..
 set(CPACK_PACKAGE_NAME "fastnetmon")
--- a/src/fastnetmon.service
+++ /dev/null
@@ -1,14 +0,0 @@
-[Unit]
-Description=FastNetMon - DoS/DDoS analyzer with sflow/netflow/mirror support
-After=syslog.target network.target remote-fs.target
- 
-[Service]
-Type=forking
-ExecStart=/opt/fastnetmon/fastnetmon --daemonize
-PIDFile=/run/fastnetmon.pid
-
-#ExecReload=/bin/kill -s HUP $MAINPID
-#ExecStop=/bin/kill -s QUIT $MAINPID
- 
-[Install]
-WantedBy=multi-user.target
--- /dev/null
+++ b/src/fastnetmon.service.in
@@ -0,0 +1,14 @@
+[Unit]
+Description=FastNetMon - DoS/DDoS analyzer with sflow/netflow/mirror support
+After=syslog.target network.target remote-fs.target
+ 
+[Service]
+Type=forking
+ExecStart=@CMAKE_INSTALL_SBINDIR@/fastnetmon --daemonize
+PIDFile=/run/fastnetmon.pid
+
+#ExecReload=/bin/kill -s HUP $MAINPID
+#ExecStop=/bin/kill -s QUIT $MAINPID
+ 
+[Install]
+WantedBy=multi-user.target
